### ISSUE TITLE
[package] opencolorio/2.2.1: broken if is_apple_os()`

### ISSUE THREAD

**Issue by tamaskenezlego** [conan-io/conan-center-index#20519]
### Description

opencolorio/2.2.1 requires its dependency minizip-ng to be built with `with_zlib=True`.
On Apple platforms, by default, minizip-ng doesn't have the `with_zlib` option because it uses `with_libcomp=True` instead. The result is this error:

```
ERROR: There are invalid packages (packages that cannot exist for this configuration):
opencolorio/2.2.1: Invalid ID: opencolorio/2.2.1 requires minizip-ng with with_zlib = True.
```
Workaround: build minizip-ng with `with_libcomp=False` and `with_zlib=True`.
Proposed fix: opencolorio/2.2.1 recipe should accept `with_libcomp=True`, as well.

### Package and Environment Details

* Package Name/Version: **opencolorio/2.2.1**

### Conan profile

[settings]
os=Macos
os_build=Macos
arch=armv8
arch_build=armv8
compiler=apple-clang
compiler.version=15
compiler.libcxx=libc++
build_type=Release
[options]
[conf]
[build_requires]
[env]

### Steps to reproduce

conan install opencolorio/2.2.1@

### Logs

<details><summary>Click to expand log</summary>

```
Configuration:
[settings]
arch=armv8
arch_build=armv8
build_type=Release
compiler=apple-clang
compiler.libcxx=libc++
compiler.version=15
os=Macos
os_build=Macos
[options]
[build_requires]
[env]

opencolorio/2.2.1: Not found in local cache, looking in remotes...
opencolorio/2.2.1: Trying with 'conancenter'...
Downloading conanmanifest.txt completed [0.37k]                                          
Downloading conanfile.py completed [7.33k]                                               
Downloading conan_export.tgz completed [0.52k]                                           
Decompressing conan_export.tgz completed [0.00k]                                         
opencolorio/2.2.1: Downloaded recipe revision 0
expat/2.5.0: Not found in local cache, looking in remotes...
expat/2.5.0: Trying with 'conancenter'...
Downloading conanmanifest.txt completed [0.10k]                                          
Downloading conanfile.py completed [3.80k]                                               
Downloading conan_export.tgz completed [0.33k]                                           
Decompressing conan_export.tgz completed [0.00k]                                         
expat/2.5.0: Downloaded recipe revision 0
openexr/3.1.7: Not found in local cache, looking in remotes...
openexr/3.1.7: Trying with 'conancenter'...
Downloading conanmanifest.txt completed [0.19k]                                          
Downloading conanfile.py completed [5.32k]                                               
Downloading conan_export.tgz completed [0.38k]                                           
Decompressing conan_export.tgz completed [0.00k]                                         
openexr/3.1.7: Downloaded recipe revision 0
WARN: openexr/3.1.7: requirement imath/3.1.9 overridden by opencolorio/2.2.1 to imath/3.1.6 
imath/3.1.6: Not found in local cache, looking in remotes...
imath/3.1.6: Trying with 'conancenter'...
Downloading conanmanifest.txt completed [0.10k]                                          
Downloading conanfile.py completed [3.72k]                                               
Downloading conan_export.tgz completed [0.33k]                                           
Decompressing conan_export.tgz completed [0.00k]                                         
imath/3.1.6: Downloaded recipe revision 0
yaml-cpp/0.7.0: Not found in local cache, looking in remotes...
yaml-cpp/0.7.0: Trying with 'conancenter'...
Downloading conanmanifest.txt completed [0.26k]                                          
Downloading conanfile.py completed [4.76k]                                               
Downloading conan_export.tgz completed [0.50k]                                           
Decompressing conan_export.tgz completed [0.00k]                                         
yaml-cpp/0.7.0: Downloaded recipe revision 0
pystring/1.1.4: Not found in local cache, looking in remotes...
pystring/1.1.4: Trying with 'conancenter'...
Downloading conanmanifest.txt completed [0.17k]                                          
Downloading conanfile.py completed [1.79k]                                               
Downloading conan_export.tgz completed [0.33k]                                           
Decompressing conan_export.tgz completed [0.00k]                                         
pystring/1.1.4: Downloaded recipe revision 0
minizip-ng/3.0.7: Not found in local cache, looking in remotes...
minizip-ng/3.0.7: Trying with 'conancenter'...
Downloading conanmanifest.txt completed [0.28k]                                          
Downloading conanfile.py completed [8.67k]                                               
Downloading conan_export.tgz completed [0.45k]                                           
Decompressing conan_export.tgz completed [0.00k]                                         
minizip-ng/3.0.7: Downloaded recipe revision 0
xz_utils/5.4.2: Not found in local cache, looking in remotes...
xz_utils/5.4.2: Trying with 'conancenter'...
Downloading conanmanifest.txt completed [0.10k]                                          
Downloading conanfile.py completed [8.58k]                                               
Downloading conan_export.tgz completed [0.30k]                                           
Decompressing conan_export.tgz completed [0.00k]                                         
xz_utils/5.4.2: Downloaded recipe revision 0
zstd/1.5.5: Not found in local cache, looking in remotes...
zstd/1.5.5: Trying with 'conancenter'...
Downloading conanmanifest.txt completed [0.20k]                                          
Downloading conanfile.py completed [3.63k]                                               
Downloading conan_export.tgz completed [0.42k]                                           
Decompressing conan_export.tgz completed [0.00k]                                         
zstd/1.5.5: Downloaded recipe revision 0
lcms/2.14: Not found in local cache, looking in remotes...
lcms/2.14: Trying with 'conancenter'...
Downloading conanmanifest.txt completed [0.20k]                                          
Downloading conanfile.py completed [3.75k]                                               
Downloading conan_export.tgz completed [0.42k]                                           
Decompressing conan_export.tgz completed [0.00k]                                         
lcms/2.14: Downloaded recipe revision 0
Version ranges solved
    Version range '>=1.2.11 <2' required by 'openexr/3.1.7' resolved to 'zlib/1.2.13' in local cache
    Version range '>=1.1 <4' required by 'minizip-ng/3.0.7' resolved to 'openssl/3.1.3' in local cache
    Version range '>=1.2.11 <2' required by 'openssl/3.1.3' resolved to 'zlib/1.2.13' in local cache

Installing package: opencolorio/2.2.1
Requirements
    bzip2/1.0.8 from 'conancenter' - Cache
    expat/2.5.0 from 'conancenter' - Downloaded
    imath/3.1.6 from 'conancenter' - Downloaded
    lcms/2.14 from 'conancenter' - Downloaded
    libiconv/1.17 from 'conancenter' - Cache
    minizip-ng/3.0.7 from 'conancenter' - Downloaded
    opencolorio/2.2.1 from 'conancenter' - Downloaded
    openexr/3.1.7 from 'conancenter' - Downloaded
    openssl/3.1.3 from 'conancenter' - Cache
    pystring/1.1.4 from 'conancenter' - Downloaded
    xz_utils/5.4.2 from 'conancenter' - Downloaded
    yaml-cpp/0.7.0 from 'conancenter' - Downloaded
    zlib/1.2.13 from 'conancenter' - Cache
    zstd/1.5.5 from 'conancenter' - Downloaded
Packages
    bzip2/1.0.8:e678d526ab22540179f155f996fc23a941dfe05a - Cache
    expat/2.5.0:c0a1a5b04ba830f6e0987bcb3d5b4ec890ecf6ea - Missing
    imath/3.1.6:c6a7b978c36bfbdf05646cef650d3b8d3c8bea46 - Missing
    lcms/2.14:6f12be38b2e454b75ce1f2efb692e5b77c8cdee3 - Missing
    libiconv/1.17:6f12be38b2e454b75ce1f2efb692e5b77c8cdee3 - Cache
    minizip-ng/3.0.7:6c60cd914bf01067d38677ee8d23d01c85e95b48 - Missing
    opencolorio/2.2.1:INVALID - Invalid
    openexr/3.1.7:91f4cdc3a7908eb76d6cc307b2b0f8ac8fd107b3 - Missing
    openssl/3.1.3:8a73dad672d6dfb98774f9915bc06ab2bd085a90 - Cache
    pystring/1.1.4:c6a7b978c36bfbdf05646cef650d3b8d3c8bea46 - Missing
    xz_utils/5.4.2:6f12be38b2e454b75ce1f2efb692e5b77c8cdee3 - Missing
    yaml-cpp/0.7.0:c6a7b978c36bfbdf05646cef650d3b8d3c8bea46 - Missing
    zlib/1.2.13:6f12be38b2e454b75ce1f2efb692e5b77c8cdee3 - Cache
    zstd/1.5.5:d8a7133d92bbb80f70995abcc81b6e067106c05c - Missing

Installing (downloading, building) binaries...
ERROR: There are invalid packages (packages that cannot exist for this configuration):
opencolorio/2.2.1: Invalid ID: opencolorio/2.2.1 requires minizip-ng with with_zlib = True.

```

</details>

**Comment by SpaceIm** [comment]
Indeed: https://github.com/conan-io/conan-center-index/pull/15669#discussion_r1287511835

**Comment by tamaskenezlego** [comment]
Proposed fix doesn't work, opened #20521 to provide a more helpful error message.
